app/src/main/java/com/zeapo/pwdstore/autofill/v2/ui/AutofillFilterView.kt

```diff
@@ -0,0 +1,157 @@
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.WindowManager
+import com.zeapo.pwdstore.R
+import kotlinx.android.synthetic.main.activity_auto_fill_filter_view.*
+import android.service.autofill.FillResponse
+import com.zeapo.pwdstore.autofill.v2.StructureParser
+import android.view.autofill.AutofillManager
+import android.app.assist.AssistStructure
+import android.content.Context
+import com.zeapo.pwdstore.autofill.v2.AutofillUtils
+import android.content.Intent
+import android.content.IntentSender
+import android.content.SharedPreferences
+import android.content.pm.ShortcutManager
+import androidx.preference.PreferenceManager
+import com.afollestad.recyclical.datasource.dataSourceOf
+import com.afollestad.recyclical.setup
+import com.afollestad.recyclical.withItem
+import com.zeapo.pwdstore.autofill.v2.ui.holder.PasswordViewHolder
+import com.zeapo.pwdstore.utils.PasswordItem
+import com.zeapo.pwdstore.utils.PasswordRepository
+import com.zeapo.pwdstore.utils.afterTextChanged
+import java.io.File
+
+
+@TargetApi(Build.VERSION_CODES.O)
+class AutofillFilterView : AppCompatActivity() {
+
+    companion object {
+        fun getFilterWindow(context: Context): IntentSender {
+            val intent = Intent(context, AutofillFilterView::class.java)
+            return PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_CANCEL_CURRENT).intentSender
+        }
+    }
+
+    private val dataSource = dataSourceOf()
+    private var settings: SharedPreferences? = null
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_auto_fill_filter_view)
+        window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+
+        settings = PreferenceManager.getDefaultSharedPreferences(this)
+
+        bindUI()
+    }
+
+    private fun bindUI() {
+        // setup{} is an extension method on RecyclerView
+        rvPassword.setup {
+            withDataSource(dataSource)
+            withItem<PasswordItem, PasswordViewHolder>(R.layout.password_row_layout) {
+                onBind(::PasswordViewHolder) { _, item ->
+                    this.label.text = item.longName
+                    this.type.text = item.fullPathToParent
+                    this.type_image.setImageResource(R.drawable.ic_action_secure)
+                }
+                onClick {
+                    setResponse(item)
+                    finish()
+                }
+            }
+        }
+
+        dataSource.set(getLastPasswordsList())
+
+        search.afterTextChanged { recursiveFilter(it, null) }
+
+        overlay.setOnClickListener {
+            setResponse(null)
+            finish()
+        }
+    }
+
+    private fun getLastPasswordsList(): List<PasswordItem> {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
+            val shortcutManager = getSystemService(ShortcutManager::class.java)
```

Comment:
`master` supports `getSystemService<ShortcutManager>()` from core-ktx extensions, use that instead.